File: Scripts/Ship.cs
using UnityEngine;

public class Ship : MonoBehaviour
{
    public string shipName;
    public string shipType;
    public int maxCrew;
    public int currentCrew;
    public float maxCargo;
    public float currentCargo;
    public float health = 100f;
    public Pirate owner;
    public bool isSelected;

    private void Start()
    {
        isSelected = false;
    }

    public bool Select()
    {
        isSelected = true;
        // Add visual indication of selection
        ShowSelectionIndicator(true);
        return true;
    }

    public void Deselect()
    {
        isSelected = false;
        // Remove visual indication of selection
        ShowSelectionIndicator(false);
    }

    private void ShowSelectionIndicator(bool show)
    {
        // Implement visual feedback for selection
        // For example, enable/disable a highlight effect or selection ring
    }

    public void SetOwner(Pirate newOwner)
    {
        owner = newOwner;
    }

    public void TakeDamage(float amount)
    {
        health = Mathf.Max(0, health - amount);
        if (health <= 0)
        {
            // Handle ship destruction
            DestroyShip();
        }
    }

    public void Repair(float amount)
    {
        health = Mathf.Min(100, health + amount);
    }

    private void DestroyShip()
    {
        // Implement ship destruction logic
        if (owner != null)
        {
            owner.RemoveShip(this);
        }
        // Add destruction effects, etc.
        Destroy(gameObject);
    }

    private void OnMouseDown()
    {
        // Handle ship selection on mouse click
        if (owner != null && owner is Player)
        {
            ((Player)owner).SelectShip(this);
        }
    }
}

File: scripts/ship.py
class Ship:
    def __init__(self, name, ship_type, max_crew, max_cargo):
        self.name = name
        self.ship_type = ship_type
        self.max_crew = max_crew
        self.current_crew = 0
        self.max_cargo = max_cargo
        self.current_cargo = 0
        self.health = 100
        self.owner = None
        self.position = (0, 0)  # X, Y coordinates on the map
        self.selected = False

    def select(self):
        """Select this ship"""
        self.selected = True
        return True

    def deselect(self):
        """Deselect this ship"""
        self.selected = False

    def set_owner(self, owner):
        """Set the owner of this ship"""
        self.owner = owner

    def move_to(self, x, y):
        """Move ship to new coordinates"""
        self.position = (x, y)

    def take_damage(self, amount):
        """Take damage and return True if ship is destroyed"""
        self.health -= amount
        return self.health <= 0

    def repair(self, amount):
        """Repair ship"""
        self.health = min(100, self.health + amount)
